### PR TITLE
Support new dos data guid

### PR DESCRIPTION
**Pull request by wshands**
Provide support for the new DOS URI format 'dg.<port number>' in the Data Object Service plugin

### CODE REVIEW

File: src/main/java/io/dockstore/provision/DOSPluginUtil.java
Comment (coverbeck) [inline src/main/java/io/dockstore/provision/DOSPluginUtil.java]: This shouldn't be part of the output.

File: src/main/java/io/dockstore/provision/DOSPluginUtil.java
Comment (coverbeck) [inline src/main/java/io/dockstore/provision/DOSPluginUtil.java]: This shouldn't be part of output.

If you want to explore using a Logger, ga4gh/dockstore#1483, you could do that and it would be useful, but if not please remove the println's.

File: src/main/java/io/dockstore/provision/DOSPluginUtil.java
Comment (coverbeck) [inline src/main/java/io/dockstore/provision/DOSPluginUtil.java]: Please remove

File: src/main/java/io/dockstore/provision/DOSPluginUtil.java
Comment (coverbeck) [inline src/main/java/io/dockstore/provision/DOSPluginUtil.java]: Please remove

File: src/main/java/io/dockstore/provision/DOSPluginUtil.java
Comment (coverbeck) [inline src/main/java/io/dockstore/provision/DOSPluginUtil.java]: Since you're listing the new format here, it might be useful to also list the old format so that the differences can easily be seen.

File: src/main/java/io/dockstore/provision/DOSPluginUtil.java
Comment (coverbeck) [inline src/main/java/io/dockstore/provision/DOSPluginUtil.java]: You're not constructing the URL here, but on line 97

File: src/main/java/io/dockstore/provision/DOSPluginUtil.java
Comment (coverbeck) [inline src/main/java/io/dockstore/provision/DOSPluginUtil.java]: It's sort of confusing `split.get(HOST)` returns a GUID. Maybe define another variable for the GUID index, which can have the same value as the HOST?

I'd suggested to Ryan that he create a class instead of using these indexes, because it gets sort of hard to read like this.

But that's probably overkill for this simple change.

**Review by coverbeck — CHANGES_REQUESTED** [review]
A lot of these comments are very nit-picky. Please do remove the printlns; the rest of the comments can be considered optional.

File: src/main/java/io/dockstore/provision/DOSPluginUtil.java
Comment (denis-yuen) [reply src/main/java/io/dockstore/provision/DOSPluginUtil.java]: Or use logging (debug or info) if possible (as time permits)

File: src/test/java/io/dockstore/provision/DOSPluginUtilUnitTest.java
Comment (denis-yuen) [inline src/test/java/io/dockstore/provision/DOSPluginUtilUnitTest.java:109]: Probably worth a comment that this is a live URL on dataguids.org and that this test (possibly others) relies on it.

File: src/main/java/io/dockstore/provision/DOSPluginUtil.java
Comment (wshands) [reply src/main/java/io/dockstore/provision/DOSPluginUtil.java]: Done

File: src/main/java/io/dockstore/provision/DOSPluginUtil.java
Comment (wshands) [reply src/main/java/io/dockstore/provision/DOSPluginUtil.java]: Removed println

File: src/main/java/io/dockstore/provision/DOSPluginUtil.java
Comment (wshands) [reply src/main/java/io/dockstore/provision/DOSPluginUtil.java]: Done

File: src/main/java/io/dockstore/provision/DOSPluginUtil.java
Comment (wshands) [reply src/main/java/io/dockstore/provision/DOSPluginUtil.java]: Removed System.out.println and added to the System.err.println

File: src/main/java/io/dockstore/provision/DOSPluginUtil.java
Comment (wshands) [reply src/main/java/io/dockstore/provision/DOSPluginUtil.java]: Added new defines

File: src/main/java/io/dockstore/provision/DOSPluginUtil.java
Comment (wshands) [reply src/main/java/io/dockstore/provision/DOSPluginUtil.java]: Removed comment

File: src/test/java/io/dockstore/provision/DOSPluginUtilUnitTest.java
Comment (wshands) [reply src/test/java/io/dockstore/provision/DOSPluginUtilUnitTest.java:109]: Added commet

**Review by coverbeck — APPROVED** [review]
Walt, please wait for Denis' approval as well before merging.

Thanks for doing this!

File: src/test/java/io/dockstore/provision/DOSPluginUtilUnitTest.java
Comment (wshands) [reply src/test/java/io/dockstore/provision/DOSPluginUtilUnitTest.java:109]: Added the comment but in a different location

**Review by denis-yuen — APPROVED** [review]

**Pull request merged**
